src/Logic.hs

```diff
@@ -130,6 +135,10 @@ runAction config action =
       doGithub $ GithubApi.leaveComment pr body
       continueWith cont
 
+    Free (IsReviewer username cont) -> do
+      hasPushAccess <- doGithub $ GithubApi.hasPushAccess username
+      continueWith $ cont hasPushAccess
```

Comment:
I have never seen Free used with pattern matching this way. Any particular reason for not using foldFree?

Reply:
Probably because I did not know what I was doing when I wrote the first version of this. I can make a new PR to clean it up later.